server/controllers/wip/questionsController.ts

```diff
@@ -1,3 +1,111 @@
+import { FormWizard } from 'hmpo-form-wizard'
+import type express from 'express'
 import { BaseController } from '../index'
+import { ReportWithDetails } from '../../data/incidentReportingApi'
+import format from '../../utils/format'
+import { getIncidentTypeConfiguration } from '../../reportConfiguration/types'
+import {
+  AnswerConfiguration,
+  IncidentTypeConfiguration,
+  QuestionConfiguration,
+} from '../../data/incidentTypeConfiguration/types'
+import logger from '../../../logger'
 
-export default class QuestionsController extends BaseController {}
+export default class QuestionsController extends BaseController<FormWizard.MultiValues> {
+  getBackLink(_req: FormWizard.Request, _res: express.Response): string {
+    return '/reports/'
+  }
+
+  middlewareLocals(): void {
+    this.use(this.lookupReport)
+    super.middlewareLocals()
+  }
+
+  async lookupReport(req: FormWizard.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+    const { incidentReportingApi } = res.locals.apis
+    const reportId = req.params.id
+
+    res.locals.report = await incidentReportingApi.getReportWithDetailsById(reportId)
+    res.locals.reportConfig = await getIncidentTypeConfiguration(res.locals.report.type)
+    next()
+  }
+
+  getValues(
+    req: FormWizard.Request<FormWizard.MultiValues, string>,
+    res: express.Response,
+    callback: FormWizard.Callback<FormWizard.MultiValues>,
+  ) {
+    return super.getValues(req, res, async (err, values) => {
+      if (err) {
+        return callback(err)
+      }
+
+      const formValues = { ...values }
+
+      const report = res.locals.report as ReportWithDetails
+      const reportConfig = res.locals.reportConfig as IncidentTypeConfiguration
+
+      for (const question of report.questions) {
+        const fieldName = this.questionIdFromCode(question.code)
+        const questionConfig: QuestionConfiguration = reportConfig.questions[fieldName]
+        if (questionConfig === undefined) {
+          logger.warn(new Error(`Question with code '${fieldName}' not found in ${report.type}'s configuration.`))
```

Comment:
can we log the report reference or id pls? and the other errors below

also, is it "just" a warning or is it an error?

Reply:
> can we log the report reference or id pls? and the other errors below
Good idea.

I started as an error tbh and then changed to a warning (as it doesn't exactly breaks) but I can try and change it back.

Reply:
Messages now include the report ID: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/269/commits/da8b69d5688c56245182a2b56952e3dc9d529cca

They're also logged as error, not just warning: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/269/commits/95640dade1117e236bef1a78c5ac2c70aaa03fe5